install.ps1

```diff
@@ -210,6 +218,63 @@ Please check you have the right permission to do so or try to create the path ma
     return (Get-Item -Path $InstallPath).FullName
 }
 
+# This function extracts the checksum of the downloaded file which contains all checksums for a version
+function Get-Checksum {
+
+    $checksumUrl = Get-ChecksumUrl
+    $arch = Get-ProcessorArchitecture
+    $response = Invoke-RestMethod -Uri $ChecksumUrl -UserAgent "PowerShell"
+    $checksums = $response -split "`n"
+    
+    $checksum = $null
+    $target = $null
+    
+    if (-not [string]::IsNullOrWhiteSpace($DaggerCommit)) {
+        $target = "dagger_${DaggerCommit}_windows_${arch}.zip"
+    }
+    else {
+        $target = "dagger_v${DaggerVersion}_windows_${arch}.zip"
+    }
+
+    # Find the checksum for the target file
+    foreach ($line in $checksums) {
+        if ($line -match $target) {
+            $checksum = $line -split " " | Select-Object -First 1
+            Write-Host "Checksum for $target is $checksum"
+            break
+        }
+    }
+
+    return $checksum
+}
+
+function Compare-Checksum {
+    Param (
+        [Parameter(Mandatory = $true)]
+        [string]$DownloadPath,
+        [Parameter(Mandatory = $true)]
+        [string]$Checksum
+    )
+
+    $hash = Get-FileHash -Path $DownloadPath -Algorithm SHA256
+
+    if ($hash.Hash -ne $Checksum) {
+        
+        Remove-Item -Path $DownloadPath
+
+        Write-Host @"
+---------------------------------------------------------------------------
+The file checksum does not match the expected checksum!
+Expected: $Checksum
+File    : $($hash.Hash)
```

Comment:
Can we add a `Got:` field as well?

Reply:
Should i rename file to Got? Since `Got` would be the same as `$hash.Hash`.

Reply:
Sorry, I misread - as-is is fine.